src/modules/job-info/guest_watch.c

```diff
@@ -763,7 +768,7 @@ static void main_namespace_lookup_continuation (flux_future_t *f, void *arg)
                                "event", tok, toklen) < 0) {
             flux_log_error (ctx->h, "%s: flux_respond_pack",
                             __FUNCTION__);
-            goto error;
+            goto cleanup;
         }
     }
 
```

Comment:
Good catch!   Though maybe the `skip_respond_error` flag could be avoided if you just moved the `error_cancel:` code into the one block that uses it and `goto cleanup` from there?